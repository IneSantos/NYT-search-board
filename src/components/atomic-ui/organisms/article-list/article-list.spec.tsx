import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import ArticlesList from "./article-list";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";

let store;
let mockedStore;

const mockedArticleList = [
  {
    id: "",
    headline: "",
    abstract: "",
    keywords: [],
    section: "",
    snippet: "",
    pub_date: "",
    source: "",
    web_url: "",
  },
];

describe("ArticlesList component tests", () => {
  test(`ArticlesList component render`, () => {
    mockedStore = configureStore();
    store = mockedStore({
      article: {
        articleList: [
          {
            id: "",
            headline: "",
            abstract: "",
            keywords: [],
            section: "",
            snippet: "",
            pub_date: "",
            source: "",
            web_url: "",
          },
        ],
        page: 0,
        searchClicked: false,
      },
    });

    const { container } = render(
      <Router>
        <Provider store={store}>
          <ArticlesList articles={mockedArticleList} />
        </Provider>
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
