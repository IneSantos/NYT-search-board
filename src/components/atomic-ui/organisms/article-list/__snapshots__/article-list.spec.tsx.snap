// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList component tests ArticlesList component render 1`] = `
<div
  class="articles-list"
>
  <h2>
    Results:
  </h2>
  <a
    href="/article/0"
    style="text-decoration: none;"
  >
    <div
      class="card-wrapper"
    >
      <div
        class="card-header"
      >
        <h3
          class="title-small"
        />
      </div>
    </div>
  </a>
</div>
`;
